File: 242-ValidAnagram/242-ValidAnagram.java
// Last updated: 5/29/2026, 11:56:48 AM
class Solution {
    public boolean isAnagram(String s, String t) {
        Map<Character,Integer> map=new HashMap<>();
        if(s.length()!=t.length()){
            return false;
        }
            for(int i=0;i<s.length();i++){
                map.put(s.charAt(i),map.getOrDefault(s.charAt(i),0)+1);
                map.put(t.charAt(i),map.getOrDefault(t.charAt(i),0)-1);
            }
            for(int count:map.values()){
                if(count!=0){
                    return false;
                }
            }
            return true;

    }
}

File: 242-ValidAnagram/242-ValidAnagram.py
# Last updated: 5/29/2026, 11:56:42 AM
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s)!=len(t):
            return False
        result={}
        for i in range(len(s)):
            result[s[i]]=result.get(s[i],0)+1
            result[t[i]]=result.get(t[i],0)-1
        for count in result.values():
            if count!=0:
                return False
        return True